datafusion/core/src/physical_plan/sorts/sort.rs

```diff
@@ -56,6 +55,27 @@ use tempfile::NamedTempFile;
 use tokio::sync::mpsc::{Receiver, Sender};
 use tokio::task;
 
+struct ExternalSorterMetrics {
```

Comment:
I agree it is much nicer to put sorting metrics on the sorter rather than BaselineMetrics